pandas/core/frame.py

```diff
@@ -7636,7 +7677,15 @@ def transform(
         assert isinstance(result, DataFrame)
         return result
 
-    def apply(self, func, axis=0, raw=False, result_type=None, args=(), **kwds):
+    def apply(
+        self,
+        func: Callable,
```

Comment:
I believe we are more specific on this, though it maybe difficult

Reply:
Looking - maybe we could do something like what was done for `AggFuncType`